// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

package feedwire;

import "components/feed/core/proto/v2/wire/chrome_client_info.proto";
import "components/feed/core/proto/v2/wire/device.proto";
import "components/feed/core/proto/v2/wire/display_info.proto";
import "components/feed/core/proto/v2/wire/version.proto";

option optimize_for = LITE_RUNTIME;

message ClientInfo {
  enum PlatformType {
    UNKNOWN_PLATFORM = 0;
    ANDROID_ID = 1;
    IOS = 2;
  }
  enum AppType {
    CHROME_ANDROID = 3;
  }
  optional PlatformType platform_type = 1;
  optional Version platform_version = 2;
  optional AppType app_type = 3;
  optional Version app_version = 4;
  optional string locale = 5;
  repeated DisplayInfo display_info = 6;
  optional string client_instance_id = 7;
  optional string advertising_id = 8;
  optional Device device = 10;
  optional ChromeClientInfo chrome_client_info = 338478298;
}
